### ISSUE TITLE
Signaling a receive task

### ISSUE THREAD

**Issue by razvancurcudel** [koolkode/bpmn#10]
Hello, 

I have a diagram with a receive task. When the execution get to this task, is setting itself in a waiting state. I have to signal it to continue the flow. 
My problem is that I don't know how to do this. 
If I use signal function from the engine->runtimeService I get 
![image](https://user-images.githubusercontent.com/9407307/40889990-04f3339c-6778-11e8-9745-6dad7147c4b9.png), and that's because, the signal function pushes a SignalVirtualExecutionCommand with signal parameter == NULL.
Only workaround that came into my mind was to call `$engine->pushCommand(new SignalVirtualExecutionCommand($executionId, "Receive message"));`, but it seems hackish to me.
Is there a nice way to do it ?

LE: I will not add this question as a separate issue.
How can I get the name of the task that waits for a signal? In the `bpmn_history_activity` table is only the id of the task
I could find the name only in the behavior of the node of a specific activity, but, there is no getter for it, and also its type is `Expression`. I couldn't figure out I could do it :-?

Thanks, 
Razvan.

**Comment by martinschroeder** [comment]
A receive task awaits a "Message" not a signal. You need to deliver a message using the `RuntimeService`.  You can use `messageEventReceived()` with the name of the message and the ID of the execution waiting within the receive task. An alternative way is to use `createMessageCorrelation()` that allows you to query for the desired execution and deliver the signal in one go.

**Comment by martinschroeder** [comment]
The answer to your second question is basically the same: use `createMessageCorrelation()` to deliver the message, you can use `activityId()` to narrow the message send operation to a specific task.

**Comment by razvancurcudel** [comment]
Thanks for answer.  
I managed to solve first problem. 

For the second problem, I think I wasn't so clear so I'll rephrase it.
For example, I have a diagram, and I want to show details about a task. One of this details is the name of task (which is shown in the diagram). I didn't figured out where I can get this field.
For user tasks is simple, because in the `bpmn_user_task` table is a field `name`, but for others there isn't such thing.

**Comment by martinschroeder** [comment]
There is indeed no way to figure out the name of a task (except for user tasks) with a query at this time. This is due to tasks not being stored in the database like user tasks are. The reason for `name` being an `Expression` is that names might contain variable substitutions that need to be evaluated in the scope of the current execution at the time the node is entered. Maybe the name could be recorded in the history of activities in a future version.

**Comment by razvancurcudel** [comment]
Oh..Ok.
Is there any chance that version will come in a very short time ?

**Comment by martinschroeder** [comment]
I added a `name` field to the activity history and adjusted events to record the names of executed activities. You can try it using `dev-master` as BPMN package version.

**Comment by razvancurcudel** [comment]
Thank you a lot. 

There is another question I want to ask. For asyncBefore tasks, implemented with Delegates classes is there a possibility to get the name of the task ? 
Node field from execution table becomes null, and in history_activity table a new record is inserted only after the task execution ends.
